gui/src/app/components/FileEditor/TextEditor.tsx

```diff
@@ -18,9 +7,27 @@ import {
   useState,
 } from "react";
 
-import monacoAddStanLang from "./monacoStanLanguage";
+import Loading from "@SpComponents/Loading";
+import { UserSettingsContext } from "@SpCore/Settings/UserSettings";
+import { CodeMarker } from "@SpCore/Stanc/Linting";
+import { unreachable } from "@SpUtil/unreachable";
+
+import { Editor, loader, useMonaco, type Monaco } from "@monaco-editor/react";
+import type { editor, IDisposable } from "monaco-editor";
+
 import { ToolBar, ToolbarItem } from "./ToolBar";
 
+import monacoAddStanLang from "./monacoStanLanguage";
+// loader from @monaco-editor/react handles the loading of the monaco editor
+// importantly, it downloads from a CDN, so we need to make sure we
+// only depend on types from the monaco-editor package to avoid
+// downloading twice.
+
+loader.config({
+  paths: {
+    vs: "https://cdn.jsdelivr.net/npm/monaco-editor@0.52.2/min/vs",
```

Comment:
Is this going to incur a maintenance cost for monaco version upgrades? (Skimming over the documentation, it isn't immediately clear to me why we need to specify a particular delivery path--isn't this the default?)

Reply:
Why is described in below comments, but yes, this does require some upkeep (though I should note pyodide has basically the same requirement in https://github.com/flatironinstitute/stan-playground/blob/main/gui/src/app/core/Scripting/pyodide/pyodideWorker.ts#L15)